Ban passing in free function into capture_pre_autograd_graph

tugsbayasgalan opened the pull request:
Summary: Today we don't allow free functions to be tracing callable in torch.export. As a part of migrating capture_preautograd_graph usages to torch.export, we need to ban free functions to capture_preautograd_graph  as well

Test Plan: CI

Differential Revision: D54319597

zhxchen17 commented on test/fx/test_matcher_utils.py:
```diff
@@ -141,11 +141,12 @@ def pattern(x, weight):
             return relu, relu_mul_by_two, {"conv": conv, "relu": relu}
 
         from torch._export import capture_pre_autograd_graph
+        from torch.export.wrapper import WrapperModule
```
@tugsbayasgalan I remember we wanna remove this wrapper, because it's hard to provide this API that works for all cases. Users probably need to wrap functions themselves.

For example, there might be a free function containing a linear layer without being registered, in this case some user would want a submodule to show up, but today we just make those tensor constants.

zhxchen17 reviewed (changes requested):
some concerns around API design.

andrewor14 commented:
Quantization changes look good to me. If you want you can dump the Wrapper in quantization as a private util API

tugsbayasgalan committed:
Ban passing in free function into capture_pre_autograd_graph

Summary: Today we don't allow free functions to be tracing callable in torch.export. As a part of migrating capture_preautograd_graph usages to torch.export, we need to ban free functions to capture_preautograd_graph  as well

Test Plan: CI

Reviewed By: zhxchen17

Differential Revision: D54319597

tugsbayasgalan commented:
@pytorchbot merge -f "Already landed"
